admin-dev/themes/new-theme/js/pages/product/combination/image-selector.ts

```diff
@@ -38,6 +38,9 @@ export default class ImageSelector {
   private init(): void {
     $(ProductMap.combinations.images.checkboxContainer, this.$selectorContainer).hide();
     this.$selectorContainer.on('click', ProductMap.combinations.images.imageChoice, (event) => {
+      if (this.$selectorContainer.hasClass('disabled')) {
+        return;
```

Comment:
I don't like having a return at the beginning of a function cause it is harder to read i think.

But in this case the function is short so it should be ok :thinking:  ...

Reply:
its considered a good practiced called "fail fast".
https://en.wikipedia.org/wiki/Fail-fast

Reply:
I didn't know this principle but taking time to read this article : https://medium.com/swlh/return-early-pattern-3d18a41bba8 I understand better your point of view I think.

Exactly the same reason I don't like exceptions, I don't like having returns everywhere and coming from C++ development, I am used to put a single return at the end.

But if you put several tests for specific cases at the beginning, I suppose it is easy to read as well.

It is funny to see that the guy has a perfect example in its article about this case in section called "Code style is subjective" xD

Thanks for your return again ^^